client/peerset/src/peersstate.rs

```diff
@@ -283,6 +283,11 @@ impl PeersState {
 		}
 	}
 
+	/// Returns `true` if there is a free outgoing slot available related to this set.
+	pub fn has_free_outgoing_slot(&self, set: usize) -> bool {
+		self.sets[set].num_out < self.sets[set].max_out
```

Comment:
```suggestion
		self.sets.get(set).map_or(false, |s| s.num_out < s.max_out)
```

Reply:
All the methods of this module panic if the set is wrong.
I'm ok with returning `false` if we also print an error and adding a `debug_assert!`, but just silently returning `false` isn't a good idea IMO.